hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/logic/ErrorReport.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.workflows.handle.flow.dispatch.logic;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.DUPLICATE_TRANSACTION;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import com.hedera.hapi.node.state.token.Account;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A report of errors that occurred during the processing of a transaction. This records the errors including the
+ * creator error, the payer error, and whether the payer was unable to pay the service fee. It also records whether
+ * the transaction is a duplicate.
+ * @param creatorId the creator account ID, should be non-null
+ * @param creatorError the creator error, can be null if there is no creator error
+ * @param payer the payer account, can be null if there is no payer account for contract operations. It can be a token address.
+ * @param payerError the final determined payer error, if any
+ * @param unableToPayServiceFee whether the payer was unable to pay the service fee
+ * @param isDuplicate whether the transaction is a duplicate
+ */
+public record ErrorReport(
```

Comment:
We probably want to specify, either in the name or the documentation, that this error is scoped to the processing of a transaction (at least, that's my understanding)

Reply:
Sure will add extra javadocs. Thanks @mhess-swl